3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -7,20 +7,24 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
 // TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+        return;
+      }
 
-    const fullName = `${firstName} Doe`;
+      const fullName = `${firstName} Doe`;
 
-    callback(fullName);
-  }, 1000);
+      resolve(fullName);
+    }, 1000);
+  });
 };
 
-getAnonName('John', console.log);
+getAnonName('John')
+  .then((message) => console.log(message))
+  .catch((err) => console.error(err));
```

Comment:
Nice job using `log` and `error`! This can be simplified a bit though
```suggestion
  .then(console.log)
  .catch(console.error);
```